pkg/controller/operators/catalog/operator.go

```diff
@@ -1444,7 +1526,7 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 		case v1alpha1.StepStatusWaitingForAPI:
 			switch step.Resource.Kind {
```

Comment:
why do we have a switch here when the `WaitingForAPI` only responds to one Kind the CRD? We don't check any other Kinds in this path. Could we maybe remove the switch and instead do a type assertion instead to make it more clear? cc @ecordell